test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/TimestampProviderTests.cs

```diff
@@ -157,12 +161,11 @@ public async Task GetTimestampAsync_WhenRequestNull_ThrowsAsync()
                     target: SignaturePlacement.PrimarySignature
                 );
 
-                // Act
-                Action timestampAction = async () => await timestampProvider.GetTimestampAsync(null, logger, CancellationToken.None);
-
                 // Assert
-                timestampAction.ShouldThrow<ArgumentNullException>()
-                    .WithMessage(string.Format(ArgumentNullExceptionMessage, nameof(request)));
+                var exception = await Assert.ThrowsAsync<ArgumentNullException>(
+                    () => timestampProvider.GetTimestampAsync(request : null, logger, CancellationToken.None));
+
+                Assert.Equal(string.Format(ArgumentNullExceptionMessage, nameof(request)), exception.Message);
```

Comment:
`nameof(request)` is incorrect here.  `nameof(request)` gets the name of the _local variable_ `request`.  The exception message contains the name of the _parameter_ `request`.  Coincidentally, the two things --- the local variable and the parameter --- have the same name but not necessarily so.  By using `nameof(request)` you are assuming/asserting that they are necessarily the same.

Using `nameof(…)` I don't think there is a way to get the parameter name from outside the method.

I would've avoided the different exception messages based on the `#if IS_DESKTOP` condition.  You could do this instead:
```C#
Assert.Equal("request", exception.ParamName);
Assert.StartsWith("Value cannot be null.", exception.Message);
```
However, the fact that the exception messages differ slightly between .NET Framework and .NET Core is unimportant to these tests.  Since these are .NET exception messages and not our own, I wouldn't bother verifying the messages themselves.  Asserting the exception type and parameter name is sufficient.
```C#
Assert.Equal("request", exception.ParamName);
```

Reply:
Thanks. Fixed.